config/recipes/logstash/logstash-multi-pipelines-as-configmaps.yaml

```diff
@@ -0,0 +1,262 @@
+      pipeline { send_to => "global_output_address" }
+    }
+
+---
+# ConfigMap for Elasticsearch output pipeline
+apiVersion: v1
+kind: ConfigMap
+metadata:
+  name: elasticsearch-output-pipeline
+data:
+  99-elasticsearch-output.conf: |
+    input {
+      # example: aggregates data from both Kafka (direct) and Beats (processed)
+      pipeline { address => "global_output_address" }
+    }
+    output {
+      # example: conditional output
+      if [pipeline_source] == "beats" {
+        elasticsearch {
+          hosts => ["${ES_HOSTS}"]
+          cacert => "${ES_SSL_CERTIFICATE_AUTHORITY}"
+          user => "${OUTPUT_ES_USER}"
+          password => "${OUTPUT_ES_PASSWORD}"
+          index => "logs-beats-default"
+        }
+      } 
+      else if [pipeline_source] == "kafka" {
+        elasticsearch {
+          hosts => ["${ES_HOSTS}"]
+          cacert => "${ES_SSL_CERTIFICATE_AUTHORITY}"
+          user => "${OUTPUT_ES_USER}"
+          password => "${OUTPUT_ES_PASSWORD}"
+          index => "logs-kafka-default"
+        }
+      }
+    }
+---
+apiVersion: logstash.k8s.elastic.co/v1alpha1
+kind: Logstash
+metadata:
+  name: multi-pipeline-logstash
+spec:
+  count: 1
+  version: 9.2.2
+  elasticsearchRefs:
+    - name: elasticsearch
+      clusterName: eck
+  pipelines:
+    # 1. Beats input
+    - pipeline.id: beats-input
+      path.config: "/etc/logstash/pipelines/beats/00-beats-input.conf"
+    # 2. Kafka input
+    - pipeline.id: kafka-input
+      path.config: "/etc/logstash/pipelines/kafka/00-kafka-input.conf"
+    # 3. Processing logic
+    - pipeline.id: processing
+      path.config: "/etc/logstash/pipelines/processing/50-processing.conf"      
+    # 4. Global output
+    - pipeline.id: global-output
+      path.config: "/etc/logstash/pipelines/output/99-elasticsearch-output.conf"
+  podTemplate:
+    spec:
+      containers:
+      - name: logstash
+        volumeMounts:
+        - name: vol-beats
+          mountPath: /etc/logstash/pipelines/beats
+          readOnly: true
+        - name: vol-kafka
+          mountPath: /etc/logstash/pipelines/kafka
+          readOnly: true
+        - name: vol-processing
+          mountPath: /etc/logstash/pipelines/processing
+          readOnly: true
+        - name: vol-output
+          mountPath: /etc/logstash/pipelines/output
+          readOnly: true        
+        env:
+        - name: OUTPUT_ES_USER
+          valueFrom:
+            secretKeyRef:
+              name: elasticsearch-credentials
+              key: es_username
+        - name: OUTPUT_ES_PASSWORD
+          valueFrom:
+            secretKeyRef:
+              name: elasticsearch-credentials
+              key: es_password
+        - name: KAFKA_BROKER_USER
+          valueFrom:
+            secretKeyRef:
+              name: kafka-pipeline-secret
+              key: kafka_username
+        - name: KAFKA_BROKER_PASSWORD
+          valueFrom:
+            secretKeyRef:
+              name: kafka-pipeline-secret
```

Comment:
🐛 This references `kafka-pipeline-secret`, but the Secret created above (line 73) is named `kafka-credentials`.